test/unit-test/mqtt_agent_utest.c

```diff
@@ -81,14 +192,196 @@ int suiteTearDown( int numFailures )
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgInterface, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+}
+
+/**
+ * @brief Test that any NULL parameter causes MQTTAgent_ResumeSession to return MQTTBadParameter.
+ */
+void test_MQTTAgent_ResumeSession_Invalid_Params( void )
 {
-    
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+
+    MQTTAgentContext_t mqttAgentContext;
+
+    setupAgentContext( &mqttAgentContext );
+
+    /* Check that MQTTBadParameter is returned if any NULL mqttAgentContext is passed. */
+    mqttStatus = MQTTAgent_ResumeSession( NULL, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* Check that MQTTBadParameter is returned if the MQTT context has not been initialized. */
+    mqttAgentContext.mqttContext.nextPacketId = 0;
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+}
+
+
+void test_MQTTAgent_ResumeSession_session_present_no_resent_publishes( void )
+{
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+
+    setupAgentContext( &mqttAgentContext );
+
+    MQTT_PublishToResend_IgnoreAndReturn( MQTT_PACKET_ID_INVALID );
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+
+void test_MQTTAgent_ResumeSession_failed_publish( void )
+{
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    Command_t command;
+    MQTTPublishInfo_t args;
+    AckInfo_t ackInfo;
+
+    setupAgentContext( &mqttAgentContext );
+
+    command.pArgs = &args;
+    ackInfo.packetId = 1U;
+    ackInfo.pOriginalCommand = &command;
+    mqttAgentContext.pPendingAcks[ 0 ] = ackInfo;
+    /* Check that failed resending publish return MQTTSendFailed. */
+    MQTT_PublishToResend_IgnoreAndReturn( 1 );
+    MQTT_Publish_IgnoreAndReturn( MQTTSendFailed );
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTSendFailed, mqttStatus );
+}
+
+void test_MQTTAgent_ResumeSession_publish_resend_success( void )
+{
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    Command_t command;
+    MQTTPublishInfo_t args;
+    AckInfo_t ackInfo;
+
+    setupAgentContext( &mqttAgentContext );
+
+    command.pArgs = &args;
+
+    ackInfo.packetId = 1U;
+    ackInfo.pOriginalCommand = &command;
+    mqttAgentContext.pPendingAcks[ 0 ] = ackInfo;
+
+    /* Check that publish ack is resent successfully when session resumes. */
+    MQTT_PublishToResend_IgnoreAndReturn( 1 );
+    MQTT_Publish_IgnoreAndReturn( MQTTSuccess );
+    MQTT_PublishToResend_IgnoreAndReturn( MQTT_PACKET_ID_INVALID );
```

Comment:
Ignore and return overrides the previous use of ignore and return, no? Shouldn't this be "ExpectAnyArgsAndReturn"?